alpine-server/src/main/java/alpine/server/auth/AuthenticationWithUriQuery.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * This file is part of Alpine.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) Steve Springett. All Rights Reserved.
+ */
+package alpine.server.auth;
+
+import jakarta.ws.rs.NameBinding;
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An annotation that indicates that a method of a JAX-RX resource is authenticated using
+ * the URI query parameter 'apiKey' instead of the header 'X-API-Key'.
+ *
+ * @author Kirill Sybin
+ * @since 3.0.2
+ */
+@NameBinding
+@Retention(value = RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+@Documented
+public @interface AuthenticationWithUriQuery {
```

Comment:
IMO a name like `AllowApiKeyInQueryParameter` would reflect the purpose better.

Reply:
Thank you very much for the review! Will do all that.

Reply:
Done